backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/deliveryplanning/DeliveryPlanningRepository.java

```diff
@@ -141,4 +147,66 @@ public void deleteForShipmentSchedule(@NonNull final ShipmentScheduleId shipment
 				.create()
 				.delete();
 	}
+
+	protected BigDecimal computePlannedLoadedQtySum(@NonNull final OrderLineId orderLineId)
+	{
+		return queryBL.createQueryBuilder(I_M_Delivery_Planning.class)
+				.addEqualsFilter(I_M_Delivery_Planning.COLUMNNAME_C_OrderLine_ID, orderLineId)
+				.create()
+				.stream()
+				.map(I_M_Delivery_Planning::getPlannedLoadedQuantity)
+				.reduce(BigDecimal.ZERO, BigDecimal::add);
```

Comment:
why not considering UOM when summing up qtys?

Reply:
also why protected?

i would avoid returning BigDecimal... but Quantity.

Reply:
I wanted to avoid having a DAO in the Repository. I have no access to the uom without UomDAO.
Modifying it by only keeping the queryBL part in here and moving the rest in the service.